src/sqlfluff/core/linter/linter.py

```diff
@@ -261,7 +262,7 @@ def _parse_tokens(
     def parse_noqa(
         comment: str,
         line_no: int,
-        rule_codes: List[str],
+        reference_map: Dict[str, List[str]],
```

Comment:
Nit: IIUC, this function only accesses the _keys_ of the dictionary. Should we consider this instead?

```
rule_codes: typing.KeysView
```

If we do this, I don't think the function itself requires any changes.

Reply:
Unfortunately we _also_ need the values of the map to be able to extract interpret any group references. This mapping is used to turn `--noqa:my_group` into `--noqa:L043,L088` (if `reference_map` is `{"my_group":{"L043", "L088"}}`).

Reply:
Oh cool. I _thought_ it must be using the values somewhere, but I overlooked it during review.